From 3a3a0b5f22e40d5dfcd837f08ed15d25c06b6042 Mon Sep 17 00:00:00 2001
From: Bastian Bittorf <bittorf@bluebottle.com>
Date: Sun, 17 Feb 2013 12:08:53 +0100
Subject: [PATCH] Failsafe-mode: print short help on commandline

Like mentioned in ticket https://dev.openwrt.org/ticket/11911
this should make the IRC much quieter. Failsafe is somehow
special and even experienced users are helpless, because they
are helpless in such a seldom situation. Also: likely you have
no internet access in this mode, so you cannot use the wiki.
---
 package/base-files/files/etc/banner.failsafe |   11 +++++++++++
 package/base-files/files/etc/profile         |    1 +
 2 files changed, 12 insertions(+), 0 deletions(-)
 create mode 100644 package/base-files/files/etc/banner.failsafe

diff --git a/package/base-files/files/etc/banner.failsafe b/package/base-files/files/etc/banner.failsafe
new file mode 100644
index 0000000..444cc6c
--- /dev/null
+++ b/package/base-files/files/etc/banner.failsafe
@@ -0,0 +1,11 @@
+================== FAILSAFE MODE ====================
+ * mount_root      mounts partition with config files
+ * /etc/config            directory with config files
+ * firstboot       reset settings to factory defaults
+ * passwd                       reset root's password
+ * reboot -f                           reboots router
+
+ please also respect:
+ http://wiki.openwrt.org/doc/howto/generic.failsafe
+=====================================================
+
diff --git a/package/base-files/files/etc/profile b/package/base-files/files/etc/profile
index 171adc1..41b8e80 100644
--- a/package/base-files/files/etc/profile
+++ b/package/base-files/files/etc/profile
@@ -1,4 +1,5 @@
 #!/bin/sh
+[ -f /tmp/.failsafe -a -f /etc/banner.failsafe ] && cat /etc/banner.failsafe
 [ -f /etc/banner ] && cat /etc/banner
 
 export PATH=/bin:/sbin:/usr/bin:/usr/sbin
-- 
1.7.2.5

